macro/src/lib.rs

```diff
@@ -69,6 +72,9 @@ pub fn load_core_bpf_program(_: TokenStream) -> TokenStream {
                     .unwrap(),
                 ),
             );
+
+            // Remove the builtin ID from the `builtins` hash set.
+            builtins.remove(&target_program_id);
```

Comment:
In the BPF implementation, the native program owners are still the NativeLoader right? In that case, you don't have to remove the builtin - we just refer to it to make sure that if the executed program is a native program, and the owner is not the native loader, then reject the input.

Reply:
The provided ELF gets inserted into the cache as a loader-v3 entry, so the program account won't align with the cache entry in the invoke context's instruction chain.

Reply:
Ah got it, that makes sense. I'll approve this change then